integreat_cms/cms/models/pages/page.py

```diff
@@ -416,17 +416,28 @@ def suggest(cls, **kwargs: Any) -> list[dict[str, Any]]:
         query = kwargs["query"]
         archived_flag = kwargs["archived_flag"]
         language_slug = kwargs["language_slug"]
+        link_suggestion_flag = kwargs["link_suggestion_flag"]
 
         pages = region.pages.all().cache_tree(archived=archived_flag)
+        title_already_in_result: list[str] = []
+
         for page in pages:
             page_translation = page.get_translation(language_slug)
-            if page_translation and (
-                query.lower() in page_translation.slug
-                or query.lower() in page_translation.title.lower()
+            if (
+                page_translation
+                and (
+                    link_suggestion_flag
+                    or page_translation.title not in title_already_in_result
+                )
+                and (
+                    query.lower() in page_translation.slug
+                    or query.lower() in page_translation.title.lower()
+                )
             ):
                 results.append(
                     format_object_translation(page_translation, "page", language_slug),
                 )
+                title_already_in_result.append(page_translation.title)
```

Comment:
If we're changing it to a set, the method changes to `add()`
```suggestion
                title_already_in_result.add(page_translation.title)
```